### PR TITLE
Fix gas energy graph units if stats added by external source

### PR DESCRIPTION
**Pull request by pszafer**
## Proposed change

By default Energy dashboard get units from entity.
If stats for gas are added by `async_add_external_statistics` then entity don't exists, so energy dashboard is falling back to m³ units.
This PR fixes this issue and get unit from statistic_meta if entity doesn't exists.

Depends on core PR https://github.com/home-assistant/core/pull/68471

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by pszafer** [comment]
`getEnergyGasUnitCategory` filters out entities to show in `Add source` dialog.
For external stats it filters out one level lower between `m³`, `ft³, `kWh`, so I think it's ok.

About correct way for filter out entities I can remove this if there is no reason to fetch units from entity itself.

File: src/data/energy.ts [hunk 1]
```diff
@@ -271,6 +278,9 @@ const getEnergyData = async (
 
   const consumptionStatIDs: string[] = [];
   const statIDs: string[] = [];
+  const statisticIdsWithMeta: StatisticsMetaData[] = await getStatisticIds(
```
Comment (balloob) [inline src/data/energy.ts:281]: it's expensive to always fetch all statistic IDs even it might not be necessary for most energy configurations.

Comment (pszafer) [reply src/data/energy.ts:281]: It's the same approach as in `statistics-chart.ts`, so I just followed it.
Anyway, there are some possibilites to change it, which do you think would be the best (or maybe other option)?
- save unit of measurement in `EnergyManager`
- add COLUMN `source type` to `statistics_meta` which could have value either `entity` or `external`, then filter based on source_type in webapi
- add filter include do webapi and do something like this
```javascript
const statisticIdsWithMeta: StatisticsMetaData[] = await getStatisticIds(
    hass
    include: prefs.energy_sources.filter(source => !(source.stat_energy_from in hass.states))
  );
```

Comment (pszafer) [reply src/data/energy.ts:281]: @balloob any thoughts about it?

Comment (pszafer) [reply src/data/energy.ts]: new WS API added https://github.com/home-assistant/core/pull/68331

**Review by emontnemery — APPROVED** [review]
Looks good to me.
This should be followed up with another PR replacing existing calls to `history/list_statistic_ids` with `recorder/get_statistics_metadata`.

**Review by zsarnett — APPROVED** [review]

**Pull request merged**